ykrt/src/compile/jitc_yk/codegen/mod.rs

```diff
@@ -0,0 +1,218 @@
+//! The JIT's Code Generator.
+
+// FIXME: eventually delete.
+#![allow(dead_code)]
+
+use super::{jit_ir, CompilationError};
+use dynasmrt::ExecutableBuffer;
+#[cfg(any(debug_assertions, test))]
+use dynasmrt::{AssemblyOffset, DynasmApi};
+#[cfg(any(debug_assertions, test))]
+use std::{cell::Cell, collections::HashMap, slice};
+use typed_index_collections::TiVec;
+
+/// FIXME: one arch is hard-compiled-in, but we should be able to "cross-unit-test" different
+/// architectures.
+#[cfg(target_arch = "x86_64")]
+mod x86_64;
+
+/// Describes storage for a local variable.
+///
+/// FIXME: This assumes everything is a stack slot. When we come to implement a register allocator,
+/// this should be an enum of stack/reg.
+#[derive(Debug, Clone, Copy)]
+struct LocalAlloc {
+    /// The offset from the base pointer of the allocation.
+    ///
+    /// This is independent of which direction the stack grows. In other words, for architectures
+    /// where the stack grows downwards, you'd subtract this from the base pointer to find the
+    /// address of the allocation.
+    ///
+    /// FIXME: when we have register allocation, consider addressing stack slots by the stack
+    /// pointer, thus freeing up the base pointer for general purpose use.
+    frame_off: usize,
+}
+
+impl LocalAlloc {
+    fn new(frame_off: usize) -> Self {
+        Self { frame_off }
+    }
+}
+
+impl LocalAlloc {
```

Comment:
Merge the `impl`s?

Reply:
Fixed in https://github.com/ykjit/yk/pull/971/commits/01814fe81a9f9325407386e5693a0fdbf764aa5c